nixos/lib/make-ext4-fs.nix

```diff
@@ -85,7 +85,11 @@ pkgs.stdenv.mkDerivation {
         size=$(( blocks - ''${free##*:} + fudge ))
 
         echo "Resizing from $blocks blocks to $size blocks. (~ $((size*blocksize/1024/1024))MiB)"
-        EXT2FS_NO_MTAB_OK=yes resize2fs $img -f $size
+        resize2fs $img -f $size || (
+          echo "Failed, resizing to minimum allowed by resize2fs"
+          fsck.ext4 -y -f $img
+          resize2fs $img -M
+        )
```

Comment:
Prefer only resize once, with `-M`, rather than fail once, then doing it with `-M`.

This also means we might be able to remove some dead code around here, which is good, no?

Reply:
I'm all for it when it comes to deleting redundant code, updated with snuggly resize dropped.